### ISSUE TITLE
assume_role does not use custom sts endpoint

### ISSUE THREAD

**Issue by hamstah** [terraform-providers/terraform-provider-aws#6424]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```provider.aws v1.43.2```

### Affected Resource(s)

Provider

### Terraform Configuration Files

```hcl
provider "aws" {
  assume_role {
    role_arn = "arn:XXXX"
  }

  endpoints {
    sts = "http://127.0.0.1:8080/aws/sts"
  }
}
```

### Debug Output

I get that message as my credentials are not valid in the AWS STS, only my custom one
```
			return nil, fmt.Errorf("The role %q cannot be assumed.\n\n"+
				"  There are a number of possible causes of this - the most common are:\n"+
				"    * The credentials used in order to assume the role are invalid\n"+
				"    * The credentials do not have appropriate permission to assume the role\n"+
				"    * The role ARN is not valid",
```

There is no API call being made to the custom endpoint

### Expected Behavior

The AssumeRole call should go to the custom endpoint

The config used for assuming the role is set at https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/auth_helpers.go#L249

Which does not set the Endpoint properly.

### Actual Behavior

The AssumeRole call goes to the AWS endpoint

### Steps to Reproduce

1. `terraform apply`

**Comment by hamstah** [comment]
@bflad hey, thanks for triaging this, is there anything else I need to do to have the fix merged in? Thanks!

**Comment by chenyosef-stratoscale** [comment]
:+1:

**Comment by bflad** [comment]
Hi folks 👋 Is this still an issue after version 2.0.0 of the Terraform AWS Provider? The authentication code was migrated to a separate library (https://github.com/hashicorp/aws-sdk-go-base) and this issue may have been addressed with some of the code refactoring during the migration.

**Comment by spellr** [comment]
:+1:

**Comment by aeschright** [comment]
I verified that this will be fixed with the aws-sdk-go-base update #10379

**Comment by bflad** [comment]
Closing as #10379 was merged previously and v2.32.0 has been released. 👍
